bin/asgs-sendmail

```diff
@@ -76,34 +78,40 @@ sub run {
         $content = q{empty message};
     }
 
-    my $config = Config::Tiny->new();
-    $config = Config::Tiny->read($config_file);
+    my $config = ini2h2o $config_file;
 
     # create log message
-    my $logmsg = sprintf(qq{subject: "%s", to: "%s"}, $subject, $to);
+    my $logmsg = sprintf( qq{subject: "%s", to: "%s"}, $subject, $to );
 
     # WAL
-    logger( type => q{WAL}, file => $logfile, msg => $logmsg);
+    logger( type => q{WAL}, file => $logfile, msg => $logmsg );
 
     my $transport = try {
-      Email::Sender::Transport::SMTP::TLS->new(
-        host     => $config->{email}->{smtp_host},
-        port     => $config->{email}->{smtp_port} // 587,    # defaults to TLS if not set
-        username => $config->{email}->{smtp_username},
-        password => $config->{email}->{smtp_password},
-        helo     => 'HELO',
-      );
+        Email::Sender::Transport::SMTP::TLS->new(
+            host     => $config->email->smtp_host,
+            port     => $config->email->smtp_port // 587,    # defaults to TLS if not set
+            username => $config->email->smtp_username,
+            password => $config->email->smtp_password,
+            helo     => 'HELO',
+        );
     }
     catch {
-      logger($logfile, qq{(FAILED) $logmsg - $_});
-      logger( type => q{FAILED}, file => $logfile, msg => qq{$logmsg - $_});
-      die "Error sending email: $_";
+        logger( $logfile, qq{(FAILED) $logmsg - $_} );
+        logger( type => q{FAILED}, file => $logfile, msg => qq{$logmsg - $_} );
+        die "Error sending email: $_";
     };
 
+    # check for subjects > 78 characters, which is the max line length according to the
+    # email RFC
+    if ( length $subject > 78 ) {
+        warn qq{(asgs-sendmail) WARNING: Length of --subject is > 78 characters, truncating ...\n};
+        $subject = substr $subject, 0, 78;
+    }
+
```

Comment:
@jasonfleming **above** is comment is the effective change, I think this is what was causing the weird issues in email in a lot of cases. If the subject line is > 78 characters, the email just doesn't make it to where it is going. I wonder how long this has been happening!

Reply:
I like this change because the email still gets through and unless it breaks on a space, it will be obvious that it is truncated, which adds to the visibility of the issue. There is a limit to the metadata we can embed in the subject line.